templates/include/app-form-required-info.html

```diff
@@ -0,0 +1,35 @@
+<!-- App Form Requred Info -->
+<h2>App Details - Required Info</h2>
+
+<!-- App Name -->
+<label for="id_name">Application Name</label>
+<input type="text" name="name" maxlength="255" required="" id="id_name" value="{{ application.name }}">
+
+<!-- App Client Type -->
+<label for="id_client_type">OAuth - Client Type</label>
+<select name="client_type" required="" id="id_client_type">
+	<option value="">---------</option>
+	{% if application.client_type == "public" %}
+	<option value="confidential">Confidential</option>
+	<option value="public" selected>Public</option>
+	{% else %}
+	<option value="confidential" selected>Confidential</option>
+	<option value="public">Public</option>
+	{% endif %}
+</select>
+
+<!-- App Grant Type -->
+<label for="id_authorization_grant_type">Authorization Grant Type*</label>
+<select name="authorization_grant_type" id="id_authorization_grant_type">
+	{% if application.authorization_grant_type == "implicit" %}
+	<option value="authorization-code">Authorization code</option>
+	<option value="implicit" selected>Implicit</option>
+	{% else %}
+	<option value="authorization-code" selected>Authorization code</option>
```

Comment:
these conditional blocks are the same.

Reply:
So - they are very nearly the same. I struggled a little bit with how to do this logic last night. The different between them is what gets the `selected` field. So, basically, we default to `authorization-code` - so I have the check to see if the app's been registered as `implicit` and select it if it has. Otherwise, either because no app's been registered or because it's been registered as `authorization-code` - that becomes the default selection.

This was one of those things where I thought there might be a better way to do it, but my brain was tired and this was what I could think of.

Reply:
got it, I missed that. That makes sense.

If you want another option, something like:
```
{% if application.authorization_grant_type == "implicit" %} selected {% endif %}
{% if application.authorization_grant_type != "implicit" %} selected {% endif %}
```
could be used. Both have their merits.

Reply:
I think I might try to implement your suggestion, since the functionality wasn't super clear on your first read-through of the code. I feel like yours highlights the `selected` toggle better.

Reply:
Implemented in a new commit.